core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -1364,6 +1365,169 @@ public void testReleaseAcquiredRecordsWithEmptyTopicPartitions() {
         assertEquals(0, result.size());
     }
 
+    @Test
+    public void testAcknowledgeSinglePartition() {
+        String groupId = "grp";
+        String memberId = Uuid.randomUuid().toString();
+
+        TopicIdPartition tp = new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("foo", 0));
+        SharePartition sp = mock(SharePartition.class);
+
+        when(sp.acknowledge(ArgumentMatchers.eq(memberId), any())).thenReturn(CompletableFuture.completedFuture(Optional.empty()));
+
+        Map<SharePartitionManager.SharePartitionKey, SharePartition> partitionCacheMap = new HashMap<>();
+        partitionCacheMap.put(new SharePartitionManager.SharePartitionKey(groupId, tp), sp);
+
+        Map<TopicIdPartition, List<ShareAcknowledgementBatch>> acknowledgeTopics = new HashMap<>();
```

Comment:
nit: You can move the declaration later in the code when the usage occurs i.e. when it's populated.